src/Concordium/Client/GRPC2.hs

```diff
@@ -0,0 +1,2766 @@
+        return QueryTypes.BlockBirkParameters{..}
+
+instance FromProto Proto.NextUpdateSequenceNumbers where
+    type Output' Proto.NextUpdateSequenceNumbers = QueryTypes.NextUpdateSequenceNumbers
+    fromProto nums = do
+        _nusnRootKeys <- fromProto =<< nums ^? ProtoFields.rootKeys
+        _nusnLevel1Keys <- fromProto =<< nums ^? ProtoFields.level1Keys
+        _nusnLevel2Keys <- fromProto =<< nums ^? ProtoFields.level2Keys
+        _nusnProtocol <- fromProto =<< nums ^? ProtoFields.protocol
+        _nusnElectionDifficulty <- fromProto =<< nums ^? ProtoFields.electionDifficulty
+        _nusnEuroPerEnergy <- fromProto =<< nums ^? ProtoFields.euroPerEnergy
+        _nusnMicroCCDPerEuro <- fromProto =<< nums ^? ProtoFields.microCcdPerEuro
+        _nusnFoundationAccount <- fromProto =<< nums ^? ProtoFields.foundationAccount
+        _nusnMintDistribution <- fromProto =<< nums ^? ProtoFields.mintDistribution
+        _nusnTransactionFeeDistribution <- fromProto =<< nums ^? ProtoFields.transactionFeeDistribution
+        _nusnGASRewards <- fromProto =<< nums ^? ProtoFields.gasRewards
+        _nusnPoolParameters <- fromProto =<< nums ^? ProtoFields.poolParameters
+        _nusnAddAnonymityRevoker <- fromProto =<< nums ^? ProtoFields.addAnonymityRevoker
+        _nusnAddIdentityProvider <- fromProto =<< nums ^? ProtoFields.addIdentityProvider
+        _nusnCooldownParameters <- fromProto =<< nums ^? ProtoFields.cooldownParameters
+        _nusnTimeParameters <- fromProto =<< nums ^? ProtoFields.timeParameters
+        return QueryTypes.NextUpdateSequenceNumbers{..}
+
+-- An IP address.
+type IpAddress = Text
+
+-- An IP port.
+type IpPort = Int16
+
+-- A peer are represented by its IP address.
+type Peer = IpAddress
+
+-- Compound type representing a pair of an IP address and a port.
+type IpSocketAddress = (IpAddress, IpPort)
+
+instance ToProto IpAddress where
+    type Output IpAddress = Proto.IpAddress
+    toProto ip = Proto.make $ ProtoFields.value .= ip
+
+instance ToProto IpPort where
+    type Output IpPort = Proto.Port
+    toProto ip = Proto.make $ ProtoFields.value .= fromIntegral ip
+
+instance FromProto Proto.IpAddress where
+    type Output' Proto.IpAddress = IpAddress
+
+    -- FIXME: Should this be validated?
+    fromProto peer = peer ^? ProtoFields.value
+
+instance FromProto Proto.Port where
+    type Output' Proto.Port = IpPort
+
+    -- FIXME: Should this be validated?
+    fromProto port = fromIntegral <$> port ^? ProtoFields.value
+
+instance FromProto Proto.BannedPeer where
+    type Output' Proto.BannedPeer = Peer
+    fromProto peer = fromProto =<< peer ^? ProtoFields.ipAddress
+
+instance FromProto Proto.BannedPeers where
+    type Output' Proto.BannedPeers = [Peer]
+    fromProto peers = mapM fromProto =<< peers ^? ProtoFields.peers
+
+instance FromProto Proto.InstanceStateValueAtKey where
+    type Output' Proto.InstanceStateValueAtKey = ByteString
+    fromProto = deMkSerialize
```

Comment:
I don't think this is correct. The value is the bytestring. It does not need to be deserialized. And if you try it will fail since bytestring serialization uses 8 bytes for length and then the actual string.